File: Samples/azure-storage/Azure.Java/models/UsageUnit.java
/**
 */

package petstore.models;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

/**
 * Defines values for UsageUnit.
 */
public enum UsageUnit {
    /** Enum value Count. */
    COUNT("Count"),

    /** Enum value Bytes. */
    BYTES("Bytes"),

    /** Enum value Seconds. */
    SECONDS("Seconds"),

    /** Enum value Percent. */
    PERCENT("Percent"),

    /** Enum value CountsPerSecond. */
    COUNTS_PER_SECOND("CountsPerSecond"),

    /** Enum value BytesPerSecond. */
    BYTES_PER_SECOND("BytesPerSecond");

    /** The actual serialized value for a UsageUnit instance. */
    private String value;

    UsageUnit(String value) {
        this.value = value;
    }

    /**
     * Parses a serialized value to a UsageUnit instance.
     *
     * @param value the serialized value to parse.
     * @return the parsed UsageUnit object, or null if unable to parse.
     */
    @JsonCreator
    public static UsageUnit fromString(String value) {
        UsageUnit[] items = UsageUnit.values();
        for (UsageUnit item : items) {
            if (item.toString().equalsIgnoreCase(value)) {
                return item;
            }
        }
        return null;
    }

    @JsonValue
    @Override
    public String toString() {
        return this.value;
    }
}

File: Samples/azure-storage/Azure.CSharp.Fluent/Models/UsageUnit.cs

namespace Petstore.Models
{

    /// <summary>
    /// Defines values for UsageUnit.
    /// </summary>
    [Newtonsoft.Json.JsonConverter(typeof(Newtonsoft.Json.Converters.StringEnumConverter))]
    public enum UsageUnit
    {
        [System.Runtime.Serialization.EnumMember(Value = "Count")]
        Count,
        [System.Runtime.Serialization.EnumMember(Value = "Bytes")]
        Bytes,
        [System.Runtime.Serialization.EnumMember(Value = "Seconds")]
        Seconds,
        [System.Runtime.Serialization.EnumMember(Value = "Percent")]
        Percent,
        [System.Runtime.Serialization.EnumMember(Value = "CountsPerSecond")]
        CountsPerSecond,
        [System.Runtime.Serialization.EnumMember(Value = "BytesPerSecond")]
        BytesPerSecond
    }
}
